### PR TITLE
[torch][ao] Properly strip tracking stats in  _fold_conv_bn_qat for 1D

### PR DESCRIPTION
**Pull request by JakeStevens**
Summary: _fold_conv_bn_qat has logic to remove the tracking stats. Currently, this includes a check that includes only torch.nn.modules.batchnorm.BatchNorm2d. As a result, the tracking stats are not properly removed when 1D is used. This diff updates to fix this.

Test Plan:
Run N7113483 without this fix.

{F1977726982}

```
bento kernel build sensorml
```

Re-run with local version of kernel, containing this diff:

{F1977727151}

Notice that now, num_batches is removed.

Differential Revision: D74269649

### CODE REVIEW

**Review by jbschlosser — COMMENTED** [review]
We'll need a test for the new behavior. I'll leave the review to the experts in this area. @yushangdi are you okay to review this?

**Comment by yushangdi** [comment]
> We'll need a test for the new behavior. I'll leave the review to the experts in this area. @yushangdi are you okay to review this?

replied in Diff. The PR looks ok to me, but would be good to get a team member from quantization team to review as well.

**Comment by yushangdi** [comment]
@JakeStevens can you add a test case in `test/quantization/pt2e/test_quantize_pt2e.py`? Thanks!

**Review by andrewor14 — APPROVED** [review]

**Comment by andrewor14** [comment]
Yes please add the test case so we know it works for sure, thanks

**Comment by JakeStevens** [comment]
@yushangdi @andrewor14 test seems flaky/failure unrelated. Is that known issue, can this be merged still? I do not have permissions.

**Review by yushangdi — APPROVED** [review]
LGTM

**Comment by yushangdi** [comment]
> @yushangdi @andrewor14 test seems flaky/failure unrelated. Is that known issue, can this be merged still? I do not have permissions.

@JakeStevens You can Ship the internal Diff, and the OSS PR will land automatically.
